lib/sqlalchemy/ext/mypy/plugin.py

```diff
@@ -217,80 +222,63 @@ def _cls_decorator_hook(ctx: ClassDefContext) -> None:
 
     assert (
         isinstance(node_type, Instance)
-        and names._type_id_for_named_node(node_type.type) is names.REGISTRY
+        and names.type_id_for_named_node(node_type.type) is names.REGISTRY
     )
 
-    decl_class._scan_declarative_assignments_and_apply_types(ctx.cls, ctx.api)
+    decl_class.scan_declarative_assignments_and_apply_types(ctx.cls, ctx.api)
 
 
 def _base_cls_decorator_hook(ctx: ClassDefContext) -> None:
     _add_globals(ctx)
 
     cls = ctx.cls
 
-    _make_declarative_meta(ctx.api, cls)
+    _set_declarative_metaclass(ctx.api, cls)
 
-    decl_class._scan_declarative_assignments_and_apply_types(
+    util.set_is_base(ctx.cls.info)
```

Comment:
**mike bayer** (zzzeek) wrote:

this is new, set_is_base

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2962